crml/nft/src/lib.rs

```diff
@@ -302,10 +340,174 @@ decl_module! {
 			royalties_schedule: Option<RoyaltiesSchedule<T::AccountId>>,
 		) {
 			let series_id = Self::next_series_id(collection_id);
-			let _ = Self::mint_series(origin, collection_id, One::one(), owner, attributes, metadata_path, royalties_schedule)?;
+			let _ = Self::mint_series(origin, collection_id, One::one(), owner, attributes, metadata_path, royalties_schedule, None)?;
 			IsSingleIssue::insert(collection_id, series_id, true);
 		}
 
+		/// Buy tokens from an active mass_drop
+		/// Checks to make sure the drop has started
+		/// If the whitelist is not empty, only whitelisted accounts will be able to join
+		#[weight = 78_000_000]
+		#[transactional]
+		fn enter_mass_drop(
+			origin,
+			collection_id: CollectionId,
+			series_id: SeriesId,
+			quantity: TokenCount,
+		) -> DispatchResult {
+			let origin = ensure_signed(origin)?;
+			let owner = Self::collection_owner(collection_id);
+			ensure!(owner.is_some(), Error::<T>::NoCollectionOwner);
+			let mass_drop: MassDrop = Self::series_mass_drops(collection_id, series_id)
+				.ok_or(Error::<T>::NoMassDropExists)?;
+			let owner = owner.unwrap();
+			let is_owner = owner == origin.clone();
+			let serial_number = Self::next_serial_number(collection_id, series_id);
+			ensure!(
+				serial_number.checked_add(quantity).is_some(),
+				Error::<T>::NoAvailableIds
+			);
+			let current_block = <frame_system::Module<T>>::block_number();
+
+			let (price, max_supply, transaction_limit) = if current_block >= T::BlockNumber::from(mass_drop.activation_time) || is_owner {
+				(mass_drop.price, mass_drop.max_supply, mass_drop.transaction_limit)
+			} else if mass_drop.presale.is_some() && current_block >= T::BlockNumber::from(mass_drop.presale.clone().unwrap().activation_time) {
+				let presale = mass_drop.presale.unwrap();
+				ensure!(
+					<PresaleWhitelist<T>>::iter_prefix((collection_id, series_id)).next().is_none() ||
+					<PresaleWhitelist<T>>::contains_key((collection_id, series_id), &origin),
+					Error::<T>::NoPermission
+				);
+				(presale.price, presale.max_supply, presale.transaction_limit)
+			} else {
+				return Err(Error::<T>::MassDropNotStarted.into());
+			};
+
+			if let Some(transaction_limit) = transaction_limit {
+				ensure!(quantity <= transaction_limit, Error::<T>::PurchaseQuantityTooHigh);
+			}
+
+			ensure!(serial_number <= max_supply, Error::<T>::NotEnoughTokens);
+			let quantity = if serial_number + quantity >= max_supply {
+				PresaleWhitelist::<T>::remove_prefix((collection_id, series_id));
+				max_supply - serial_number
+			}else {
+				quantity
+			};
+
+			Self::process_drop_payment(&origin, owner, price, quantity, mass_drop.asset_id)?;
+			Self::do_mint_additional(&origin, collection_id, series_id, serial_number, quantity)?;
+			Self::deposit_event(RawEvent::EnterMassDrop(collection_id, series_id, quantity, origin));
```

Comment:
please add the first serial number in this event
```suggestion
			Self::deposit_event(RawEvent::EnterMassDrop(collection_id, series_id, quantity, origin));
```